engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -606,29 +530,20 @@ public void saveComponent(long entityId, Component component) {
      * Implementation
      */
 
-    protected void assignToPool(EntityRef ref, EngineEntityPool pool) {
-        //Todo: job for the sector manager?
-        poolMap.put(ref.getId(), pool);
+    private EngineEntityPool getEntityPool(long id) {
+        EngineEntityPool pool = poolMap.get(id);
+        if (pool == null) {
+                logger.error("Entity {} doesn't have an assigned pool", id);
+                return globalPool;
+        } else {
+            return pool;
+        }
     }
 
     protected void assignToPool(long entityId, EngineEntityPool pool) {
```

Comment:
Hmm, I need to think about that... It probably should so that system can react on this change. It might be especially interesting if some systems wants to collect some resources or information just _before_ the entity gets shifted from chunk scope to sector scope.